engine/lib/luajit-ffi-gen/src/impl_item/generate_rust.rs

```diff
@@ -262,48 +300,116 @@ impl ImplInfo {
                 let ty_ident = format_ident!("{}", ty.variant.as_ffi().rust);
 
                 if ty.wrapper == TypeWrapper::Option {
-                    quote! { *const #ty_ident }
+                    if ty.is_mutable {
+                        quote! { Option<&mut #ty_ident> }
+                    } else if ty.is_reference {
+                        quote! { Option<&#ty_ident> }
+                    } else {
+                        // We pin an instance of Option<T> using gen_buffered_ret, so ensure the
+                        // right lifetime is used here.
+                        quote! { Option<&'static #ty_ident> }
+                    }
                 } else {
                     quote! { #ty_ident }
                 }
             }
         }
     }
 
-    fn gen_wrapper_param_to_impl(
+    fn gen_wrapper_name_from_ffi(
         &self,
-        param: &ParamInfo,
+        name: &String,
+        ty: &TypeInfo,
         use_convert_into_string: &mut bool,
     ) -> TokenStream {
-        let name_ident = format_ident!("{}", param.name);
-        let name_accessor =
-            if param.ty.wrapper == TypeWrapper::Option && !param.ty.variant.is_string() {
-                quote! { (*#name_ident) }
-            } else {
-                quote! { #name_ident }
-            };
+        let name_ident = format_ident!("{}", name);
+        let name_accessor = quote! { #name_ident };
 
-        let param_item = match &param.ty.variant {
+        match &ty.variant {
+            TypeVariant::Function { args, ret } => {
+                // Assign a name for each argument.
+                let args: Vec<_> = args
+                    .iter()
+                    .enumerate()
+                    .map(|(index, ty)| (format!("arg{}", index), ty))
+                    .collect();
+
+                // Generate the tokens for the argument list.
+                let arg_tokens: Vec<_> = args
+                    .iter()
+                    .map(|(name, _)| format_ident!("{}", name))
+                    .collect();
+
+                // Convert the Rust expressions to FFI.
+                let mut prelude: Vec<TokenStream> = vec![];
```

Comment:
I think you don't need `Vec<TokenStream>` here, compiler should deduct it by itself.